src/test/librados/service.cc

```diff
@@ -45,6 +46,69 @@ TEST(LibRadosService, RegisterLate) {
   rados_shutdown(cluster);
 }
 
+static void status_format_func(const int i)
+{
+  rados_t cluster;
+  ASSERT_EQ(0, rados_create(&cluster, "admin"));
+  ASSERT_EQ(0, rados_conf_read_file(cluster, NULL));
+  ASSERT_EQ(0, rados_conf_parse_env(cluster, NULL));
+
+  ASSERT_EQ(0, rados_connect(cluster));
+  string tid = stringify(std::this_thread::get_id());
+  string name = string("portal/gw") + stringify(i % 8) + string(":rbd/image") + tid;
+  ASSERT_EQ(0, rados_service_register(cluster, "iscsi", name.c_str(),
+                                      "foo\0bar\0this\0that\0"));
+  sleep(20);
```

Comment:
why?

Reply:
Without this the threads will be exit immediately, which will unregister the services too.

We need to make all the threads to keep running for a while, or the `ceph -s` won't see the 128 services at the same time when checking them in `StatusFormat` later.

Reply:
Yeah, this approach sounds cool, will try it later.